DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1

```diff
@@ -32,35 +35,56 @@ function Get-TargetResource
          }
     }
 
-    # Mount VHD.
+
+    # Mount VHD
     $mountVHD = EnsureVHDState -Mounted -vhdPath $vhdPath
-    
-    $itemsFound = foreach($Item in $FileDirectory)
+
+    try 
     {
-        $item = GetItemToCopy -item $item
-        $mountedDrive =  $mountVHD | Get-Disk | Get-Partition | Get-Volume
-        $letterDrive  = "$($mountedDrive.DriveLetter):\" 
-       
-        # show the drive letters.
-        Get-PSDrive | Write-Verbose       
-
-        $finalPath = Join-Path $letterDrive $item.DestinationPath
+        # Show the drive letters after mount 
+        Write-Verbose "List of PSDrives follow"
+        Get-PSDrive | Write-Verbose
+        Write-Verbose "List of all partitions (selectable with PartitionNumber parameter)"
+        $mountedVHD | Get-Disk | Get-Partition | foreach{"PartitionNumber: {0}; DriveLetter: {1}; Size: {2}; Type: {3}" -f $_.PartitionNumber,$_.DriveLetter, $_.Size, $_.Type} | Write-Verbose
         
-        Write-Verbose "Getting the current value at $finalPath ..."
 
-        if (Test-Path $finalPath)
+
+        $mountedPartition = $mountedVHD | Get-Disk | Get-Partition | Where{$_.PartitionNumber -like $PartitionNumber}
+        if(!$mountedPartition)
         {
-            New-CimInstance -ClassName MSFT_FileDirectoryConfiguration -Property @{DestinationPath = $finalPath; Ensure = "Present"} -Namespace root/microsoft/windows/desiredstateconfiguration -ClientOnly
+            throw "Partition not found"
```

Comment:
During Get-TargetResource. we don't normally throw errors.  you just need to return a CIM instance that indicate the destination path is missing. See the existing cod.